scipy/sparse/lil.py

```diff
@@ -111,6 +115,29 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                 for i in range(M):
                     self.rows[i] = []
                     self.data[i] = []
+            elif len(arg1) == 2:
+                # (data, rows) format
+                (data, rows) = arg1
+                rows_len = len(rows)
+
+                if len(data) != rows_len:
+                    raise ValueError('data and rows should have same length')
+                
+                if shape is None:
+                    raise ValueError('shape parameter is required with (data, rows) format')
```

Comment:
There's a precedent for inferring the shape if one isn't provided explicitly (see the COO constructor).

In this case, it would be something like:

```
shape = (rows_len, max(max(r) for r in rows) + 1)
```

Reply:
I will update the docstring and test also.